### PR TITLE
Enhance Basic Auth

### LINKED ISSUE #5440: Basic Auth does not use Panel favicon or color
I'm on the current Panel `main` branch. I wanted to try out basic auth.

```
panel serve script.py --basic-auth my_password --cookie-secret my_super_safe_cookie_secret
```

But it does not use the Panel favicon or blue color for the Button. This makes it look like a half finished solution.

Personally I don't understand why you want to expose a link to what for the users of your app is complicated documentation of auth. That is just confusing and information overload.

image

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Closes #5440

Addresses #5411

You can preview the updated docs and form [here](https://github.com/holoviz/panel/blob/fix/basic-auth/doc/how_to/authentication/basic.md)

### CODE REVIEW

File: panel/auth.py [hunk 1]
```diff
@@ -863,7 +863,7 @@ def post(self):
             self.set_current_user(username)
             self.redirect("/")
         else:
-            error_msg = "?error=" + tornado.escape.url_escape("Login incorrect")
+            error_msg = "?error=" + tornado.escape.url_escape("Invalid username or password!")
```
Comment (MarcSkovMadsen) [inline panel/auth.py:866]: I believe this is the text you normally see.

File: doc/how_to/authentication/basic.md [hunk 2]
```diff
@@ -45,32 +69,4 @@ When serving an application dynamically using `pn.serve` you can also provide a
...
-    <!--Login Button-->
-    <div class="form-group">
-        <button class="form-button" type="submit">Login</button>
-    </div>
-    <div><small>
-    <p><a href="https://panel.holoviz.org/how_to/authentication/index.html">See the documentation</a> for a full discussion.</p>
-    </small></div>
-</form>
-```
+If you want to customize the authentication template you can provide a custom template with the `--basic-login-template` CLI argument. The template needs to submit `username` and `password` to the `/login` endpoint of the Panel server, e.g. the form of the default template looks like this. Check out the default template [here](https://github.com/holoviz/panel/blob/main/panel/_templates/basic_login.html) for inspiration.
```
Comment (MarcSkovMadsen) [inline doc/how_to/authentication/basic.md:72]: It helps me more to see the entire template than a part of it. And it also requires work to keep the subset here updated. So replaced it with a link instead.

File: panel/_templates/basic_login.html [hunk 3]
```diff
@@ -2,7 +2,11 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>Panel App with BasicAuth</title>
+    <meta charset="utf-8">
+    <meta content="width=device-width, initial-scale=1.0" name="viewport">
+    <meta content="en" name="docsearch:language">
+    <title>Panel App | Login</title>
+    <link rel="icon" type="image/x-icon" href="https://panel.holoviz.org/_static/favicon.ico">
```
Comment (MarcSkovMadsen) [inline panel/_templates/basic_login.html:9]: I tried `href="_static/favicon.ico"`, but it did not find the favicon.

File: panel/_templates/basic_login.html [hunk 4]
```diff
@@ -2,7 +2,11 @@
 <!DOCTYPE html>
 <html>
 <head>
```
Comment (MarcSkovMadsen) [inline panel/_templates/basic_login.html:4]: A use does not care about the type of authentication applied. But its nice to know you are on a login page.

File: panel/_templates/basic_login.html (hunk 3 above)
Comment (MarcSkovMadsen) [reply panel/_templates/basic_login.html:9]: Can you fix this @philippjfr ?

**Comment by MarcSkovMadsen** [comment]
I will also try to fix issue with `/` vs `./` when Panel is not served at `/`. For example when serving in the terminal of a Jupyter Hub that with `jupyter-server-proxy`.

**Comment by MarcSkovMadsen** [comment]
Solving the Jupyterhub issue is a seperate issue. I've described it in #5446.

**Comment by philippjfr** [comment]
Merging this for now.

**Review by philippjfr — APPROVED** [review]
Looks great, thank you!

**Pull request merged**